csrc/tensor_metadata.cpp

```diff
@@ -37,24 +37,32 @@ class ForwardTraverseFromLogicalToAlloc {
       // TODO: see [Allocation domain on both side of logical]
       return;
     }
+
     auto [in_size, in_stride] = in_it->second;
     auto factor = ee_.evaluate(split->factor()).as<int64_t>();
     NVF_ERROR(
         in_size % factor == 0,
         "The logical domain and allocation domain of fusion input/output ",
         "tensors must be a one-to-one map, therefore, ",
         "non-divisible split is not allowed in allocation domain");
+
+    int64_t inner_size;
+    int64_t outer_size;
+    if (split->innerSplit()) {
```

Comment:
@jjsjann123 this check was missing. I believe there's a similar problem in BackwardTraverseFromLogicalToAlloc which I didn't fix in this PR.

Reply:
I think we need to have a test exposing that first before fixing anything.

Reply:
Yeah no worries. We'll patch it when it shows up.